x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -490,6 +495,39 @@ private LogicalPlan resolveMvExpand(MvExpand p, List<Attribute> childrenOutput)
             return p;
         }
 
+        private LogicalPlan resolveLookup(Lookup l, List<Attribute> childrenOutput) {
+            // check if the table exists before performing any resolution
+            if (l.localRelation() == null) {
+                return l;
+            }
+
+            // check the on field against both the child output and the inner relation
+            List<NamedExpression> on = l.matchFields();
+            List<NamedExpression> resolvedOn = new ArrayList<>(on.size());
+            List<Attribute> localOutput = l.localRelation().output();
+
+            for (NamedExpression ne : on) {
+                if (ne instanceof UnresolvedAttribute ua && ua.customMessage() == false) {
+                    ne = maybeResolveAttribute(ua, localOutput);
+                    // can't find the field inside the local relation
+                    if (ne instanceof UnresolvedAttribute lua) {
+                        // adjust message
+                        ne = lua.withUnresolvedMessage(
+                            lua.unresolvedMessage().replace("Unknown column", "Unknown column in lookup target")
+                        );
+                    }
+                    // check also the child output by resolving to it
+                    else {
+                        ne = maybeResolveAttribute(ua, localOutput);
+                    }
+                }
+
+                resolvedOn.add(ne);
+            }
```

Comment:
Just a long winded way of doing resolution for better error messages (missing in the target lookup table vs the child/main table).